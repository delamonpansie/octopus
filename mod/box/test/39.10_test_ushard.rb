#!/usr/bin/ruby
# coding: utf-8

$: << File.dirname($0)
require '39_test_ushard'

$one_env.meta 'shard 1 create por'
$one_env.meta 'shard 1 add_replica three'

$one_env.meta 'shard 2 create por'
$one_env.meta 'shard 2 add_replica two'

$two_env.meta 'shard 3 create por'
$two_env.meta 'shard 3 add_replica three'

$one.create_object_space 0, :shard => 1, :index => DEFIDX
$one.insert [1,"one"], :shard => 1

$one.create_object_space 0, :shard => 2, :index => DEFIDX
$one.insert [1,"one"], :shard => 2

$one.create_object_space 0, :shard => 3, :index => DEFIDX
$one.insert [1,"one"], :shard => 3

$three = ThreeEnv.new.connect_eval do
  self.connect_name = "three"
  ping
  self
end

sleep 0.1

$three.select 1, :shard => 1
$three.select 1, :shard => 3
