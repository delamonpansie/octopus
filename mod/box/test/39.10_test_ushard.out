# one.ping()
:pong

# two.ping()
:pong

# one.meta(shard 1 create por)
ok

# one.meta(shard 1 add_replica three)
ok

# one.meta(shard 2 create por)
ok

# one.meta(shard 2 add_replica two)
ok

# two.meta(shard 3 create por)
ok

# two.meta(shard 3 add_replica three)
ok

# one.create_object_space(0, {:shard=>1, :index=>{:type=>:FASTTREE, :unique=>1, :field_0=>{:type=>:STRING, :index=>0, :sort_order=>:DESC}}})
:success

# one.insert([1, "one"], {:shard=>1})
1

# one.create_object_space(0, {:shard=>2, :index=>{:type=>:FASTTREE, :unique=>1, :field_0=>{:type=>:STRING, :index=>0, :sort_order=>:DESC}}})
:success

# one.insert([1, "one"], {:shard=>2})
1

# one.create_object_space(0, {:shard=>3, :index=>{:type=>:FASTTREE, :unique=>1, :field_0=>{:type=>:STRING, :index=>0, :sort_order=>:DESC}}})
:success

# one.insert([1, "one"], {:shard=>3})
1

# three.ping()
:pong

# three.select(1, {:shard=>1})
[["\x01\x00\x00\x00", "one"]]

# three.select(1, {:shard=>3})
[["\x01\x00\x00\x00", "one"]]

